### PR TITLE
[LLVM] Update ARM maintainers

### PR DESCRIPTION
**Pull request by nikic**
@rengolin You are currently listed as the maintainer for the ARM backend, but I think nowadays you mostly work on the MLIR side and aren't active in the ARM backend anymore, right?

I'd like to nominate @davemgreen as the new maintainer for ARM.

### CODE REVIEW

**Review by rengolin — APPROVED** [review]
Ah, yes! It has been a while. Happy with the change, thanks!

**Comment by davemgreen** [comment]
Cool. Hmm... I'm not sure if it is very useful to have just me in the list, I'm just trying to help. In reality a lot of people have been helping out with the Arm backend for a long time and it would be a shame if we disrupted that. I would consider putting "Arms compiler team" as the first member on the list, but I can see that's not always very useful to people. I would suggest adding @ostannard for the assembly/disassembly, as he knows a lot more about that area than me. @eefriedman has done a lot of work over the years, I'm not sure if he would want the be on the list. We should find someone who looks after target parser too. The arm abi I would add @smithp35 to but that seems to be a separate section. @nasherm mentioned he was interested recently too.

Can we expand this a little to something like this, although maybe the "Arm's Compiler Team" could do with a better explanation.

```
#### ARM backend

Arm's Compiler Team \
David Green  david.green@arm.com (email), [davemgreen](https://github.com/davemgreen) (GitHub) \
Oliver Stannard  oliver.stannard@arm.com (email), [ostannard](https://github.com/ostannard) (GitHub) \
Nashe Mncube  nashe.mncube@arm.com (email), [nasherm](https://github.com/nasherm) (GitHub)
```

**Review by ostannard — APPROVED** [review]
I think @davemgreen is a good choice for this. I'd also be happy to put my name down as a second maintainer.

**Comment by nasherm** [comment]
I also agree @davemgreen is a good choice. I'm happy to be on this list as a tertiary maintainer.

**Review by nasherm — APPROVED** [review]

**Comment by nikic** [comment]
@davemgreen Thanks for the response! I should have probably linked to https://llvm.org/docs/DeveloperPolicy.html#maintainers to clarify what the expectations for maintainers are.

> I would consider putting "Arms compiler team" as the first member on the list, but I can see that's not always very useful to people.

The most important requirement here is really that there is some way to contact maintainers -- I think we could add something like "Arms compiler team" if there is at least a group email address (though a pingable team on GitHub would be ideal).

> I would suggest adding @ostannard for the assembly/disassembly, as he knows a lot more about that area than me. @eefriedman has done a lot of work over the years, I'm not sure if he would want the be on the list. We should find someone who looks after target parser too. The arm abi I would add @smithp35 to but that seems to be a separate section. @nasherm mentioned he was interested recently too.

Happy to add additional people here, and we can also split this up further, no need to limit to a generic "ARM backend" category. Could do something like this:

```
#### ARM backend

David Green \
david.green@arm.com (email), [davemgreen](https://github.com/davemgreen) (GitHub)

#### ARM backend - assembler/disassembler

Oliver Stannard \
oliver.stannard@arm.com (email), [ostannard](https://github.com/ostannard) (GitHub)

#### ARM backend - ABI

Peter Smith \
peter.smith@arm.com (email), [smithp35](https://github.com/smithp35) (GitHub)
```

etc.

Related to this @asl you are currently listed as the maintainer for ARM EABI at https://github.com/llvm/llvm-project/blob/main/llvm/Maintainers.md#arm-eabi. Is that still up to date?

**Comment by asl** [comment]
> Related to this @asl you are currently listed as the maintainer for ARM EABI at https://github.com/llvm/llvm-project/blob/main/llvm/Maintainers.md#arm-eabi. Is that still up to date?

I think ARM EABI support is done for many years as of now. And given that these days we are having ARM compiler team contributing, it seems natural to move everything ARM-related to them :)

**Comment by davemgreen** [comment]
> The most important requirement here is really that there is some way to contact maintainers -- I think we could add something like "Arms compiler team" if there is at least a group email address (though a pingable team on GitHub would be ideal).

Yeah I agree, I was thinking of something like the old PowerPC group from phabricator, but I don't know of a suitable alias or how well it works in practice. I would just want to encourage everyone to get involved and help out, not feeling like they have to leave it to others.

> Could do something like this:

Sounds good. In fear of adding too many to the list, @stuij also offered to help with early architecture support and global-isel, which is an area that could do with the attention so would be good to add. I would also like to add Peter for ABI (as a whole, not EABI). I'm not sure how best it is to structure, something like this might be OK for now if it is acceptable, we can always change it later. If this isn't the best (I don't know how it renders) then either one big section or several should be OK.
```
David Green \
david.green@arm.com (email), [davemgreen](https://github.com/davemgreen) (GitHub) \
Oliver Stannard (Especially assembly/dissassembly) \
oliver.stannard@arm.com (email), [ostannard](https://github.com/ostannard) (GitHub) \
Nashe Mncube \
nashe.mncube@arm.com (email), [nasherm](https://github.com/nasherm) (GitHub) \
Peter Smith (Anything ABI) \
peter.smith@arm.com (email), [smithp35](https://github.com/smithp35) (GitHub) \
Ties Stuij (GlobalISel and early arch support) \
ties.stuij@arm.com (email), [stuij](https://github.com/stuij) (GitHub) \
```

**Comment by nikic** [comment]
@davemgreen Thanks, I updated the PR with your suggested list. I like the more compact format you suggested.

**Review by asl — APPROVED** [review]

**Review by davemgreen — APPROVED** [review]
Thanks. LGTM

**Review by ostannard — APPROVED** [review]

**Review by smithp35 — APPROVED** [review]
LGTM too

**Review by stuij — APPROVED** [review]
👍

**Review by nasherm — APPROVED** [review]
LGTM

**Pull request merged**
